### PR TITLE
Add a completion handlers for pushing/presenting

### PR DESCRIPTION
**Pull request by samirGuerdah**
Attempt to address the issue #5

### CODE REVIEW

File: Sources/RouterServiceInterface/PresentationStyle.swift [hunk 1]
```diff
@@ -18,14 +19,18 @@ open class PushPresentationStyle: PresentationStyle {
     open func present(
         viewController: UIViewController,
         fromViewController: UIViewController,
-        animated: Bool
+        animated: Bool,
+        completion:(() -> Void)?
     ) {
+        CATransaction.begin()
+        CATransaction.setCompletionBlock(completion)
```
Comment (bruno-hcr) [inline Sources/RouterServiceInterface/PresentationStyle.swift:26]: What do you think to abstract these `CATransaction` methods and present with completion from **PresentationStyle**?
One suggestion is creating a commitTransaction free function
```swift
func commitTransaction(_ transaction: () -> Void, completion: () -> Void) {
    CATransaction.begin()
    CATransaction.setCompletionBlock(completion)
    transaction()
    CATransaction.commit()
}
```

And another suggestion is to create a UINavigationController extension for `pushViewController(_:)` method. In that way, you can implement future other methods and implementations like `pop(_:)`, `popToRoot(_:)`, etc.

```swift
extension UINavigationController {
    func pushViewController(_ viewController: UIViewController, animated: Bool, completion: @escaping () -> Void) {
        commitTransaction {
            pushViewController(viewController, animated: animated)
        }, completion: completion
    }
}
```

Comment (samirGuerdah) [reply Sources/RouterServiceInterface/PresentationStyle.swift:26]: I like your suggestion, I am just wondering where to put the free function and the extension method. 
PresentationStyle source file is not the right place. Any suggestion :) ?

Comment (rockbruno) [reply Sources/RouterServiceInterface/PresentationStyle.swift:26]: We can put the free function inside the same file as a private one, as I think we won't need this for somewhere else

Comment (rockbruno) [reply Sources/RouterServiceInterface/PresentationStyle.swift:26]: BTW, we'll also need public extensions for the PresentationStyle to add methods that don't take a completion handler, just so we don't break previous implementations.

**Comment by bruno-hcr** [comment]
@samirGuerdah is there any updates about this PR?

**Comment by bruno-hcr** [comment]
PR will be closed by discontinuity!

**Pull request closed without merging**